From c2b88c3e09701c331067d0c2f90ecc2d351c5ca7 Mon Sep 17 00:00:00 2001
From: "hjnim.kim" <hjnim.kim@samsung.com>
Date: Mon, 7 Oct 2013 16:35:24 +0900
Subject: [PATCH] JQM Popup resize event handler is called until reisze event
 is ended

JQM Popup resize event handler is called until resize event is ended
N_SE-49435

Signed-off-by : hyunjung kim <hjnim.kim@samsung.com>

Change-Id: I625b20d0283cefe13c2eead7dd3c11781145fb5c
---
 libs/js/jquery-mobile-1.2.0/js/widgets/popup.js |   12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/libs/js/jquery-mobile-1.2.0/js/widgets/popup.js b/libs/js/jquery-mobile-1.2.0/js/widgets/popup.js
index 3508291..0d6aee2 100644
--- a/libs/js/jquery-mobile-1.2.0/js/widgets/popup.js
+++ b/libs/js/jquery-mobile-1.2.0/js/widgets/popup.js
@@ -113,7 +113,7 @@ define( [ "jquery",
 		},
 
 		_resizeTimeout: function() {
-			if ( !this._maybeRefreshTimeout() && this.positionTo === "window" && this._isOpen ) {
+			if ( this.positionTo === "window" && this._isOpen ) {
 				// effectively rapid-open the popup while leaving the screen intact
 				this._trigger( "beforeposition" );
 				this._ui.container
@@ -123,6 +123,8 @@ define( [ "jquery",
 				this._resizeScreen();
 				this._resizeData = null;
 				this._orientationchangeInProgress = false;
+			} else if( !this._isOpen && this._isPreOpen  )  {
+				setTimeout( $.proxy( this, "_resizeTimeout" ), 200 );
 			}
 		},
 
@@ -135,6 +137,10 @@ define( [ "jquery",
 				}
 				this._maybeRefreshTimeout();
 			}
+
+			if ( !this._isOpen && this._isPreOpen && this.positionTo === "window" ) {
+				 setTimeout( $.proxy( this, "_resizeTimeout" ), 200 );
+			}
 		},
 
 		_handleWindowOrientationchange: function( e ) {
@@ -198,6 +204,7 @@ define( [ "jquery",
 				_fallbackTransition: "",
 				_currentTransition: false,
 				_prereqs: null,
+				_isPreOpen: false,
 				_isOpen: false,
 				_tolerance: null,
 				_resizeData: null,
@@ -619,6 +626,9 @@ define( [ "jquery",
 					return false;
 				}());
 
+			// If popup is received resize event during the orientationchange ( resize ) event, resize handler is called once more
+			this._isPreOpen = true;
+
 			// Make sure options is defined
 			options = ( options || {} );
 
-- 
1.7.9.5

